libs/wingsdk/src/shared-gcp/function.inflight.ts

```diff
@@ -0,0 +1,90 @@
+import { GoogleAuth } from "google-auth-library";
+import { IFunctionClient } from "../cloud";
+import { RequestRedirect, Util as http } from "../http";
+import { Trace } from "../std";
+
+export class FunctionClient implements IFunctionClient {
+  constructor(
+    private readonly functionName: string,
+    private readonly projectId: string,
+    private readonly region: string
+  ) {}
+
+  private async _invokeLocally(
+    payload: string,
+    token: string
+  ): Promise<string> {
+    const url = `https://${this.region}-${this.projectId}.cloudfunctions.net/${this.functionName}`;
+    try {
+      const res = await http.post(url, {
+        headers: {
+          body: payload,
+          "Content-Type": "text/plain",
+          authorization: `bearer ${token}`,
+        },
+        redirect: RequestRedirect.FOLLOW,
+      });
+      if (!res.ok) {
+        throw new Error(res.body);
+      }
+      return res.body ?? "";
+    } catch (error) {
+      throw new Error(
+        `Error while invoking the function ${this.functionName}:\n${
+          (error as Error).message
+        }`
+      );
+    }
+  }
+
+  /**
+   * Invoke the function, passing the given payload as an argument.
+   *  @returns the function returned payload only
+   */
+  public async invoke(payload: string): Promise<string> {
+    const url = `https://${this.region}-${this.projectId}.cloudfunctions.net/${this.functionName}`;
+    try {
+      const auth = new GoogleAuth();
+      const client = await auth.getIdTokenClient(url);
+
```

Comment:
Im curious, not sure I see it in the code (maybe its a GCP thing) but what permission does the function require in order to make this `auth.getIdTokenClient(url)`?